client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -120,19 +122,31 @@ module.exports = React.createClass( {
 
 			if ( cartItems.hasRenewalItem( this.state.previousCart ) ) {
 				renewalItem = cartItems.getRenewalItems( this.state.previousCart )[ 0 ];
+
 				redirectTo = purchasePaths.managePurchase( renewalItem.extra.purchaseDomain, renewalItem.extra.purchaseId );
 			}
 		}
 
 		page.redirect( redirectTo );
+
 		return true;
 	},
 
 	getCheckoutCompleteRedirectPath: function() {
 		var renewalItem;
+
 		if ( cartItems.hasRenewalItem( this.props.cart ) ) {
 			renewalItem = cartItems.getRenewalItems( this.props.cart )[ 0 ];
+
 			return purchasePaths.managePurchaseDestination( renewalItem.extra.purchaseDomain, renewalItem.extra.purchaseId, 'thank-you' );
+		} else if ( cartItems.hasFreeTrial( this.props.cart ) ) {
```

Comment:
I'm not a big fan of this change. I feel like it doesn't improve the clarity of the code and handles some edge cases in a function which should simply return a path.
In the case of Paypal we don't care about stale data since the whole app is going to reload anyway. So we can ignore this case.
I think it would make sense to me that the code responsible for making the payment request would itself trigger the invalidation (not necessarily handle it but notify of a possible modification in the site data).
For instance this could be handled in SecurePaymentForm.componentWillReceiveProps, once this.props.transaction.step.name === 'received-wpcom-response'